scripts/storage-facts.py

```diff
@@ -205,7 +205,7 @@ def generateFacts(original_facts, storage_host):
             attributes['reservation'] = facts[
               'vm_facts_default_root_reservation'] if 'vm_facts_default_root_reservation' in facts else config['size']
         # Don't share via NFS on backup hosts
-        attributes['sharenfs'] = 'off' if not nfs_options_to_set else ','.join(sorted(set(nfs_options_to_set)))
+        attributes['sharenfs'] = 'off' if not nfs_options_to_set else ','.join(sorted(set(nfs_options_to_set), key=lambda x: nfs_options_to_set.index(x)))
```

Comment:
Why not just `join(nfs_options_to_set)`? I think you sort it with a key that keeps the original order.

Reply:
Because we want to get rid of duplicates